### PR TITLE
Updated the code to generate blocks for the given address.

### PR DESCRIPTION
**Pull request by surangap**
#### What kind of PR is this?:

/kind fix

#### What this PR does / why we need it:
will generate to the given address. currently, it is generated to the first masternode, not to the passed one.
#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -190,15 +190,29 @@ static UniValue generatetoaddress(const JSONRPCRequest& request)
...
+    }
+
+    CKeyID myID;
+    // check myID is in myAllMNs
+    for (const auto& mnId : myAllMNs) {
+      if (mnId.first == passedID) {
+        myID = mnId.first;
+        break;
+      }
+    }
```
Comment (bvbfan) [inline src/rpc/mining.cpp:208]: Just use `count`
```
if (myAllMNs.count(passedID)) {
    myID = passedID;
} else {
    myID = myAllMNs.begin()->first;
}
```

**Review by bvbfan — CHANGES_REQUESTED** [review]

**Review by bvbfan — APPROVED** [review]

File: src/rpc/mining.cpp [hunk 2]
```diff
@@ -37,6 +37,8 @@
 
 #include <memory>
 #include <stdint.h>
+#include <set>
+#include <utility>
```
Comment (ShengguangXiao) [inline src/rpc/mining.cpp:41]: These two includes have no use

File: test/functional/rpc_getmininginfo.py [hunk 3]
```diff
@@ -32,16 +33,17 @@ def run_test(self):
         operators = [node0_keys.operatorAuthAddress, node1_keys.operatorAuthAddress]
 
         self.log.info("Restart nodes...")
-        self.restart_node(0, ['-gen', '-masternode_operator=' + operators[0]])
-        self.restart_node(1, ['-gen', '-rewardaddress=' + operators[1]] +
+        self.restart_node(0, ['-gen', '-dummypos=0', '-masternode_operator=' + operators[0]])
+        self.restart_node(1, ['-gen','-dummypos=0', '-rewardaddress=' + operators[1]] +
```
Comment (ShengguangXiao) [inline test/functional/rpc_getmininginfo.py:37]: Add blank after ,

**Comment by ShengguangXiao** [comment]
The logic looks good, i just left two minor comments

**Review by ShengguangXiao — DISMISSED** [review]

**Review by bvbfan — DISMISSED** [review]

File: src/rpc/mining.cpp [hunk 4]
```diff
@@ -190,13 +190,24 @@ static UniValue generatetoaddress(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Error: Invalid address");
     }
 
-    auto myIDs = pcustomcsview->AmIOperator();
-    if (!myIDs) {
-        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Error: I am not masternode operator");
+    CKeyID passedID = destination.which() == PKHashType ? CKeyID(*boost::get<PKHash>(&destination)) :
```
Comment (prasannavl) [inline src/rpc/mining.cpp:193]: Could you please break this line into a switch statement instead? That would be far more readable and way easier to maintain. Everything else looks good.

**Review by prasannavl — CHANGES_REQUESTED** [review]
Just the one minor nit pick, and it's good to merge. Thanks!

**Comment by prasannavl** [comment]
Perfect. Looks good now, thanks!

**Review by prasannavl — APPROVED** [review]

**Pull request merged**